src/__tests__/element-queries.js

```diff
@@ -565,4 +565,63 @@ test('getByText ignores script tags by default', () => {
   expect(getAllByText(/hello/i, {ignore: false})).toHaveLength(3)
 })
 
+test('get/query input element by current value', () => {
+  const {
+    getByCurrentValue,
+    queryByCurrentValue,
+    getByTestId,
+  } = renderIntoDocument(`
+    <div>
+      <input placeholder="name" type="text" data-testid="name" value="Mercury" />
+    </div>
+  `)
+  expect(getByCurrentValue('Mercury').placeholder).toEqual('name')
+  expect(queryByCurrentValue('Mercury').placeholder).toEqual('name')
+
+  getByTestId('name').value = 'Norris'
+  expect(getByCurrentValue('Norris').placeholder).toEqual('name')
+  expect(queryByCurrentValue('Norris').placeholder).toEqual('name')
+})
+
+test('get/query select element by current value', () => {
+  const {
+    getByCurrentValue,
+    queryByCurrentValue,
+    getByTestId,
+  } = renderIntoDocument(`
+    <select id="state-select" data-testid="state">
+      <option value="">State</option>
+      <option value="AL">Alabama</option>
+      <option selected value="AK" >Alaska</option>
+      <option value="AZ">Arizona</option>
+    </select>
+  `)
+
+  expect(getByCurrentValue('Alaska').id).toEqual('state-select')
+  expect(queryByCurrentValue('Alaska').id).toEqual('state-select')
+
+  getByTestId('state').value = 'AL'
+  expect(getByCurrentValue('Alabama').id).toEqual('state-select')
```

Comment:
I wonder if it's at all confusing that we are expected to set the `value` to `AL` but when we use the query we search for `Alabama`.

I realize that's just how things work (the value for a select has a display value), but I feel like this could be a point of confusion for people. My guess is they'd prefer to select by `AL` instead which isn't what is shown to the user but feels more intuitive here. I'm not sure of the best solution to this... Maybe allow for either the display value or the actual value?

Reply:
That makes sense. To end-users `Alabama` is a value, but `AL` is the actual value.

Reply:
However, `selectElem.value` could be either `1`, `AL`, or whatever user cannot know. In a perspective of writing tests as user would use it, I'd rather choose to restrict it to display value.
What do you think?

Reply:
Yeah, I think `displayValue` is probably a better mental model, though it may be a tiny confusing for `input` and `textarea` where the _display_ value is actually the _actual_ value but I think that's alright. Besides, people should normally be getting these elements by their label anyway, so I'm not too concerned about this slightly less-awesome API.

Let's go with `getByDisplayValue` 👍